packages/@aws-cdk/aws-codedeploy/lib/server/deployment-group.ts

```diff
@@ -348,6 +348,7 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupBase {
     switch (asg.osType) {
       case ec2.OperatingSystemType.LINUX:
         asg.addUserData(
+          'set +e', // make sure we don't exit on command errors - we check the return codes below
```

Comment:
Feels like this should be in a subshell, or should be reversed afterwards (but a subshell feels safer). It's risky to switch off a global setting without switching it back.

Reply:
@rix0rrr I didn't like the way the sub-shell solution worked, so I did something different - I minimized the places where we use `set +e`, and made the script work with `set -e` (which I think is the correct thing to do).

Let me know if this works!